generate-stats

```diff
@@ -100,6 +102,34 @@ def main():
     # Sort tables
     all_tables = sorted(all_tables, key=lambda x: x["data"][0][1], reverse=True)
 
+    # Select unexpected messages
+    messages = cursor.execute("""\
+            SELECT retry_reason, TestRuns.url
+            FROM Failures
+            JOIN TestRuns ON Failures.run = TestRuns.id
+            WHERE TestRuns.time > ? AND
+                retry_reason IS NOT null AND
+                retry_reason IS NOT 'be robust against unstable tests' AND
+                retry_reason IS NOT 'due to failure of test harness or framework';""", (since,)).fetchall()
+
+    messages = [m for m in messages if m[0] is not None]
+    # Replace any numbers > 100 with 0 (usually time stamps or PIDs)
+    messages = [(re.sub(r'\d\d[\d\.]+', '0', m[0]), m[1]) for m in messages]
+
+    # Replace any partial paths like dracut.xf8hG with dracut.xxx
+    messages = [(re.sub(r'\/([^\/]*)\.[^\/]*\/', r'/\1.xxx/', m[0]), m[1]) for m in messages]
+
+    data = []
+    for c in Counter([m[0] for m in messages]).most_common():
+        if (c[1] > 1):  # Don't show message that happened just once
```

Comment:
Good call! I think after we establish some percentages for the flaking tests, we should also not show trivial ones (below the "error budget"), so that the page will turn into a proper TODO list.